### ISSUE TITLE
`history -c` should be more verbose to prevent misuse

### ISSUE THREAD

**Issue by Sargates** [ohmyzsh/ohmyzsh#12472]
### Describe the proposed feature.

Using `history -c` should be more verbose. I saw that the command was mentioned in patch notes (I believe from closing #12362 because the timeframe matches up) so naturally I ran the command because I didn't know what it did. 
Then I realized that my entire history file was deleted, which was full of useful commands that I can't remember the syntax for.
My point is the command to delete the history file should be more verbose than just the `-c` flag. Something like `--delete` would be more blatant and prevent accidents like this from happening to others.

**Comment by mcornella** [comment]
`history -c` is a well-known command from clearing the history, coming from the bash background. Changing the flag will not be a backwards-compatible change, but we can indeed make it safer by adding a confirmation prompt.

**Comment by mcornella** [comment]
Fixed in 35a6725970167278254ab11a996bf04d2186b009, `history -c` will now ask for confirmation before proceeding. You can get the update with `omz update`. Thanks for the idea!

**Comment by robots4life** [comment]
How can I send a `y` to `history -c` in this alias `alias hit='history -c && exit'` with this change please ?

With this change so far now I have to confirm every time I run the alias, however I would like to not have to do that manually, thank you.

**Comment by mcornella** [comment]
You can pass in `yes | history -c` to the command, so the alias would be
```sh
alias hit='yes | history -c && exit' 
```

**Comment by robots4life** [comment]
With `yes` in place it still asks for confirmation.

```shell
alias hit='yes | history -c && exit'
```

**Comment by mcornella** [comment]
Oh should be `yes y` sorry, was on mobile and couldn't verify.

**Comment by emakman** [comment]
`yes` should have been correct, and `yes y` isn't working. History doesn't seem to take the piped input at all.

**Comment by robots4life** [comment]
Yeah seems this does not work either.

Would it be possible to either revert this change so I can run the alias as before or have a setting in the `.zshrc` file that could turn this particular warning off ?

**Comment by mcornella** [comment]
Fixed in c83ca51b1b2fce89c5eae0b0bc9600ee4df28e08. `read -k1` was not reading from `stdin`, causing the issue. Either way, if this is something that you're always doing, you might want to look at disabling the history file entirely via
```sh
HISTFILE=
```
in your .zshrc file, after Oh My Zsh is sourced, to turn history off.

**Comment by robots4life** [comment]
Thank you.

I am not always doing this but thank you for letting me know that this would also be possible as a permanent solution with the `.zshrc` config and the `HISTFILE=` option.

What is the proper way of passing `yes` or `y` to `history` in the alias please ?

```shell
alias hit='yes y | history -c'
```

or

```shell
alias hit='yes | history -c'
```

?

**Comment by rwmitchell** [comment]
You could just try running ```yes```, ```yes y```, and ```yes n``` to see what they do.

**Comment by robots4life** [comment]
@rwmitchell 

> You could just try running `yes`, `yes y`, and `yes n` to see what they do. You'll find the first 2 are the same and the third will bring enlightenment.

I asked since I did indeed ran them both and found out that  `yes` and `yes y` are the same, so I wanted to know if having the one over the other has deeper implications with regards to omz or `history`.

@mcornella 

For the `n` option, not sure if this is intended, now when passing `yes n` there is a `%` instead of the passed `n` at the end of the prompt.

```shell
alias abc123='yes n | history -c'
```
This is the output.

```shell
This action will irreversibly delete your command history. Are you sure? [y/N] %
```

Passing `yes n` I would expect to see a `n` at the end of the prompt.

**Comment by mcornella** [comment]
Any of these work. As explained in https://github.com/ohmyzsh/ohmyzsh/issues/12472#issuecomment-2170969182, `yes` already outputs a `y` so it is more straightforward to use `yes |`.

**Comment by mcornella** [comment]
> For the `n` option, not sure if this is intended, now when passing `yes n` there is a `%` instead of the passed `n` at the end of the prompt.
> 
> ```shell
> alias abc123='yes n | history -c'
> ```
> 
> This is the output.
> 
> ```shell
> This action will irreversibly delete your command history. Are you sure? [y/N] %
> ```
> 
> Passing `yes n` I would expect to see a `n` at the end of the prompt.

For this (which also happens with `yes |`) I pushed f2769acdfa283c663bb1bf2b98890766622e410c.

**Comment by rwmitchell** [comment]
For clarification I didn't intend my tests to be used in the actual alias command but to be run on the command line to understand what the ```yes``` command does.  Once that is understood, it should be clear that getting a 'y' as input, whether from ```yes y``` or just ```yes``` doesn't make a difference, both are y

**Comment by robots4life** [comment]
> For clarification I didn't intend my tests to be used in the actual alias command but to be run on the command line to understand what the `yes` command does.

I did test 

```shell
yes | history -c
```

vs.

```shell
yes y | history -c
```

vs.

having both variations in an alias.

Before this commit https://github.com/ohmyzsh/ohmyzsh/commit/f2769acdfa283c663bb1bf2b98890766622e410c there was a difference to alias vs. command, I was just not 100% sure if it would other implications.

Thank you for the fix and the clarifications. :heart:

**Comment by rwmitchell** [comment]
My example didn't show piping the output anywhere, it was intended to go to the terminal for you to learn what the ```yes``` command outputs and why ```yes`` vs ```yes y``` makes no difference except one is longer.

**Comment by robots4life** [comment]
@rwmitchell 

The context of my question https://github.com/ohmyzsh/ohmyzsh/issues/12472#issuecomment-2169273207 is given and clear, the context is the alias with omz. Given this, I indeed understood your response that way but also did try many combinations of the `yes` command, perhaps that was not clear enough on my part. Either way there must have been a misunderstanding somewhere between my question and your initial response, hence thank you for keeping at it. Given the rest of the replies on this issue I was also wondering if there would be any other implications with omz, regardless of what the `yes` command does and how it works.

Bottom line, I am indeed super glad you do keep following up on it despite the matter being resolved. Glad this is all clear now but if you like, be my guest, keep adding more helpful responses. One never knows, could be one can come up with yet even more helpful info on the topic.
